src/Sylius/Component/Core/Model/Order.php

```diff
@@ -527,13 +527,54 @@ public function getPromotions()
     }
 
     /**
+     * Returns sum of neutral and non neutral tax adjustments on order and total tax of order items.
+     *
+     * {@inheritdoc}
+     */
+    public function getTaxTotal()
+    {
+        $taxTotal = 0;
+
+        foreach ($this->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT) as $taxAdjustment) {
+            $taxTotal += $taxAdjustment->getAmount();
+        }
+        foreach ($this->items as $item) {
+            $taxTotal += $item->getTaxTotal();
+        }
+
+        return $taxTotal;
+    }
+
+    /**
+     * Returns shipping fee together with taxes decreased by shipping discount.
+     *
      * @return int
      */
-    public function getPromotionsTotalRecursively()
+    public function getShippingTotal()
     {
-        return
-            $this->getAdjustmentsTotalRecursively(AdjustmentInterface::ORDER_PROMOTION_ADJUSTMENT) +
-            $this->getAdjustmentsTotalRecursively(AdjustmentInterface::ORDER_ITEM_PROMOTION_ADJUSTMENT)
-        ;
+        $shippingTotal = $this->getAdjustmentsTotal(AdjustmentInterface::TAX_ADJUSTMENT);
+        $shippingTotal += $this->getAdjustmentsTotal(AdjustmentInterface::ORDER_PROMOTION_ADJUSTMENT);
+
+        foreach ($this->getAdjustments(AdjustmentInterface::SHIPPING_ADJUSTMENT) as $shippingAdjustment) {
+            $shippingTotal += $shippingAdjustment->getAmount();
+        }
+
+        return $shippingTotal;
+    }
+
+    /**
+     * Returns amount of order discount. Does not include order item and shipping discounts.
+     *
+     * @return int
+     */
+    public function getOrderPromotionTotal()
+    {
+        $orderPromotionTotal = 0;
+
+        foreach ($this->items as $item) {
+            $orderPromotionTotal += $item->getAdjustmentsTotalRecursively(AdjustmentInterface::ORDER_PROMOTION_ADJUSTMENT);
```

Comment:
Why would you need to introduce a `SHIPPING_TAX_ADJUSTMENT`? There's no requirement for this. Ideally there should be no requirement for `SHIPPING_PROMOTION_ADJUSTMENT` either, it should all just be done as a `PROMOTION_ADJUSTMENT` that is applied to the relevant item (`Order`, `OrderItem`, `OrderItemUnit`), but you needed to separate out Adjustment types for logic internal to promotions.

There are many use cases where you want to know how much discount there has been on shipping, whether it's on a single order, part of a group of orders, a request for total discount or just shipping discounts.

It's more about consistency. If all other promotional discounts are represented by adjustment types, why would you add a special business rule for shipping discounts which is "look for all negative SHIPPING_ADJUSTMENTS". It just feels like a bad design.

In our implementation, and I'm guessing most people's now, the only adjustments that can really be on an `Order` object are shipping adjustments anyway. So any TAX_ADJUSTMENT, or X_PROMOTION_ADJUSTMENT at this level would be for shipping - since all other adjustments are split down to `OrderItemUnits`.